pkg/machines/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -489,10 +523,18 @@ class CreateVmModal extends React.Component {
             );
             this.setState({ [key]: value });
             break;
-        case 'storageSize':
+        case 'storageSize': {
+            const storagePools = this.props.storagePools.filter(pool => pool.connectionName === this.state.connectionName);
```

Comment:
This should be tested as well. That choosing bigger value than the available size will round it down. Currently I have some tests which check for errors in case too big storage size is selected.